packages/app-builder/src/models/webhook.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  type WebhookDto,
+  type WebhookRegisterBodyDto,
+  type WebhookSecretDto,
+  type WebhookUpdateBodyDto,
+  type WebhookWithSecretDto,
+} from 'marble-api/generated/marblecore-api';
+import invariant from 'tiny-invariant';
+
+export const eventTypes = ['case_status_updated'] as const;
+type EventType = (typeof eventTypes)[number];
```

Comment:
can you please explain this syntax to me ? 🤔

Reply:
This is TS syntax: basically it create a type from all the elements of the "enumerable" provided type (here, union of each `eventTypes` items type).

Since we use a `as const` declaration, TS don't infer  :
```TS
const eventTypes: string[]
type EventType = string
```
But
```TS
const eventTypes: readonly ["case_status_updated"]
type EventType = "case_status_updated"
```

Reply:
thanks,
yes that was my intuition, but why the `[number]` ?

Reply:
I don't know 🤷🏽 
Probably the syntax decision in TS to tell you want to "get the value of the array type by iterating with any number" (since indexes are numbers)